pkg/external_repos/introspect.go

```diff
@@ -51,6 +52,18 @@ func Introspect(repo dao.PublicRepository, rpm dao.RpmDao) (int64, error) {
 	if client, err = httpClient(IsRedHat(repo.URL)); err != nil {
 		return 0, err
 	}
+
+	revision, _ := yum.GetRevision(client, repo.URL)
+	if repo.Revision != "" && revision != "" && revision == repo.Revision {
+		// If repository hasn't changed, no need to update
+		return 0, nil
+	}
+
+	repo.Revision = revision
+	if err = repoDao.UpdateRepository(repo); err != nil {
+		return 0, err
+	}
```

Comment:
we should save the revision only if the introspection completes successfully

Reply:
Good point